### PR TITLE
ActionExecutingContext.Result is not set after invoking next

### LINKED ISSUE #27647: After the next filter is executed and the result is changed, the _actionExecutingContext.Result is still null.
I used multiple filters, the following filter set result in OnActionExecuted action, when execute the OnActionExecution action of previous filter again, context.Result is still null.

### PR DESCRIPTION
**Pull request by ilonze**
After the next filter is executed and the result is changed, the _actionExecutingContext.Result is still null. 
I used multiple filters, the following filter set result in OnActionExecuted action, when execute the OnActionExecution action of previous filter again, context.Result is still null.

Fixes #27647

### CODE REVIEW

**Review by enisn — APPROVED** [review]

**Comment by enisn** [comment]
It looks right in theory

**Comment by ilonze** [comment]
Hi!
It looks like merging is still blocked... need a review and someone with write access to merge this!
Thanks!

**Comment by davidfowl** [comment]
Where's the test?

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/aspnetcore?pullRequest=27639) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/aspnetcore?pullRequest=27639) before we can accept your contribution.<br/><br/>:x: ilonze [sign now](https://cla.dotnetfoundation.org/dotnet/aspnetcore?pullRequest=27639)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/aspnetcore?pullRequest=27639) it.</sub>

**Comment by davidfowl** [comment]
@ilonze Thanks! Can you fix up the PR comment? Is there a bug number for this?

**Comment by ilonze** [comment]
@davidfowl Sorry! I haven't bug number.

**Comment by davidfowl** [comment]
@ilonze Can you fix up the pull request comment as well?

**Comment by ilonze** [comment]
@davidfowl Sorry! I'm edited.

File: src/Mvc/Mvc.Core/src/Infrastructure/ControllerActionInvoker.cs [hunk 1]
```diff
@@ -91,6 +91,10 @@ private Task Next(ref State next, ref Scope scope, ref object state, ref bool is
                             {
                                 _actionExecutingContext = new ActionExecutingContextSealed(_controllerContext, _filters, _arguments, _instance);
                             }
+							else if (_actionExecutedContext != null && _actionExecutedContext.Result != null)
```
Comment (JamesNK) [inline src/Mvc/Mvc.Core/src/Infrastructure/ControllerActionInvoker.cs:94]: Replace tabs with spaces

File: src/Mvc/Mvc.Core/src/Infrastructure/ControllerActionInvoker.cs [hunk 2]
```diff
@@ -101,6 +105,10 @@ private Task Next(ref State next, ref Scope scope, ref object state, ref bool is
...
```
Comment (JamesNK) [inline src/Mvc/Mvc.Core/src/Infrastructure/ControllerActionInvoker.cs:108]: Replace tabs with spaces

**Review by enisn — APPROVED** [review]

**Review by pranavkm — CHANGES_REQUESTED** [review]
Hi, I asked some clarifying questions about this PR in the issue: https://github.com/dotnet/aspnetcore/issues/27647#issuecomment-725583408. Could you respond to it?

**Comment by ilonze** [comment]
> Hi, I asked some clarifying questions about this PR in the issue: [#27647 (comment)](https://github.com/dotnet/aspnetcore/issues/27647#issuecomment-725583408). Could you respond to it?

Hi, My demo added to the issue: [#27647 (comment)](https://github.com/dotnet/aspnetcore/issues/27647#issuecomment-725784804)

**Review by pranavkm — APPROVED** [review]

**Review by javiercn — APPROVED** [review]

File: src/Mvc/Mvc.Core/src/Infrastructure/ControllerActionInvoker.cs [hunk 3]
```diff
@@ -91,6 +91,10 @@ private Task Next(ref State next, ref Scope scope, ref object state, ref bool is
                             {
                                 _actionExecutingContext = new ActionExecutingContextSealed(_controllerContext, _filters, _arguments, _instance);
                             }
+                            else if (_actionExecutedContext != null && _actionExecutedContext.Result != null)
```
Comment (davidfowl) [inline src/Mvc/Mvc.Core/src/Infrastructure/ControllerActionInvoker.cs:94]: nit: could be simplified to:

```C#
if (_actionExecutedContext?.Result != null)
```

Comment (javiercn) [reply src/Mvc/Mvc.Core/src/Infrastructure/ControllerActionInvoker.cs:94]: 💯 Pranav points :)

**Comment by pranavkm** [comment]
@ilonze sorry for taking this long to get back to your PR. It looks like some of the tests in MVC are failing as a result of this change. Could you look?

**Comment by pranavkm** [comment]
@ilonze feel free to reopen this PR once you've had a chance to look at the test failures.

**Pull request closed without merging**